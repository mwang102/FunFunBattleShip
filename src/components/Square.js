import React from 'react'
import styled from "styled-components";
import PropTypes from 'prop-types'

    /* 2 flipped and hit
       1 not flipped yet and hit 
       0 not flipped and nothing under
      -1 flipped and nothing under 
    */

export default class Square extends React.Component{
  static propTypes = {
    rowNumber:PropTypes.number.isRequired,
    columnNumber:PropTypes.number.isRequired,
    data:PropTypes.array.isRequired
  }

  render(){
    const { data, rowNumber, columnNumber } = this.props
    const currentSquare = {
        row:rowNumber, 
        column:columnNumber, 
        data:data
        },
        squareBackground = data[0] > 0 && data[1] === true ? 'rgb(' + (80 + (data[0] * 50)) + ',0,0)' : 
                           data[0] ===  -1 ? 'SteelBlue' : 'BlanchedAlmond',
        hoverBackground = data.squareValue === 2 || data.squareValue === -1 ? '' : 'gold'

    return(
      <SquareContainer hoverBackground={hoverBackground} backGround = {squareBackground} squareValue={currentSquare.data}
        onClick={()=> this.props.handlePick(currentSquare)}>
      </SquareContainer>
    )
  }
}

const SquareContainer = styled.div`
    display:flex;
    justify-content:center;
    align-items:center;
    margin: 0px 5px;
    flex: 1 0 auto;
    padding: 0px 1.0875rem 1.45rem;
    box-shadow: 0 1px 5px 0 rgba(0,0,0,0.5);
    font-size:30px;
    background-repeat:no-repeat;
    background-position: center; 
    background:${props => props.backGround };
    cursor:${props => props.squareValue[1] ? '' : 'pointer'}};
    -webkit-animation:${props => props.squareValue[1] ? 'flipInX 1s' : ''}};
    :hover{
      background:${props => props.hoverBackground }
`;
