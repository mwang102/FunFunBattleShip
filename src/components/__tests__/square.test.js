import React from 'react';
import { shallow } from 'enzyme'
import Square from '../Square'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

describe('square component',()=>{
	it('checks if there is one square', ()=>{
		const mockData = [5,false],
			  mockCallback = jest.fn()

	    const wrapper = shallow(<Square rowNumber data={mockData} rowNumber={0} columnNumber={0}/>)
		expect(wrapper.length).toEqual(1)
	})

	it('Square color when not flipped and future miss',()=>{
		const mockData = [0,false],
			  mockCallback = jest.fn()

		const wrapper = shallow(<Square handlePick={mockCallback} data={mockData} rowNumber={0} columnNumber={0}/>)
		expect(wrapper).toHaveStyleRule('background', 'BlanchedAlmond')
	})

	it('Square color when flipped and hit',()=>{
		const mockData = [5,true],
			  mockCallback = jest.fn()

		const wrapper = shallow(<Square handlePick={mockCallback} data={mockData} rowNumber={0} columnNumber={0}/>)
		expect(wrapper).toHaveStyleRule('background', 'rgb(' + (80 + (mockData[0] * 50)) + ',0,0)' )
	})

	it('Square color when flipped and miss',()=>{
		const mockData = [-1,true],
			  mockCallback = jest.fn()

		const wrapper = shallow(<Square handlePick={mockCallback} data={mockData} rowNumber={0} columnNumber={0}/>)
		expect(wrapper).toHaveStyleRule('background', 'SteelBlue' )
	})

	it('matches the snapshot', ()=>{
		const mockData = [0,false],
			  mockCallback = jest.fn()

		const tree = renderer.create(<Square handlePick={mockCallback} data={mockData} rowNumber={0} columnNumber={0}/>).toJSON()
		
		expect(tree).toMatchSnapshot()
	})

})