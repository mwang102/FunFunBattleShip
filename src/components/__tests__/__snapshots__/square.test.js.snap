// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`square component matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px 5px;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding: 0px 1.0875rem 1.45rem;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.5);
  font-size: 30px;
  background-repeat: no-repeat;
  background-position: center;
  background: BlanchedAlmond;
  cursor: pointer;
}

.c0:hover {
  background: gold;
}

<div
  className="c0"
  onClick={[Function]}
/>
`;
